libraries/node-core-library/src/JsonSchema.ts

```diff
@@ -210,6 +215,18 @@ export class JsonSchema {
     }
   }
 
+  /**
+   * Helper function to determine if the targeted schema is using the draft-04 specification
+   * @returns boolean value indicating that the schema uses the draft-04 specification
+   */
+  private _isDraft04Schema(): boolean {
+    if (this._schemaObject.$schema) {
+      return this._schemaObject.$schema.startsWith('http://json-schema.org/draft-04/');
+    } else {
+      return false;
+    }
+  }
```

Comment:
If someone doesn't have a `$schema` property, we're defaulting to draft-07?

Can we include an option to the initializers' `options` object to explicitly specify which version of the schema the 
consumer wants?

Reply:
yes the `ajv` defaults to the json-schema version that is linked to the provided 'ajv' module export, which defaults  to draft-07. There are alternative module exports available for draft-04, draft-2019 and draft-2020.

We can add an option to force a specific json-schema version, but the downside is that we would be introducing a secondary mechanism to provide a schema version next to the schema itself. We could also see conflicts if mismatched versions are provided, so I'm not sure if we gain a lot by doing so...

Reply:
I think it'd be useful to be able to explicitly dictate which version of the schema you want to target. It'd also be useful to create an API to dictate the version (either with an enum or a string union) to create precedent for supporting future versions.

Reply:
👍 I provided an initial version of expose a schema version option, let me know what you think

Reply:
Pushed a few tweaks.